rust/noosphere-fs/src/fs.rs

```diff
@@ -277,7 +277,12 @@ where
     ) -> Result<Cid> {
         let sphere = Sphere::at(&self.sphere_revision, &self.db);
 
-        // TODO(#85): Error result if mutation is empty and also additional headers are empty
+        let mutation_to_save = self.mutation.get().is_some() && !self.mutation.get().expect("").is_empty();
```

Comment:
Since `mutation` is a `OnceCell`, there is a chance that it has not been initialized. There is a helper function on this struct that we use two lines below (L287) to get an error result in case the `mutation` isn't initialized.

We should move that up into a `let` binding so that we only do the check once:

```rs
let mutation = self.require_mutation()?;
```

Then you can avoid writing `self.mutation.get()` multiple times.